core/src/test/scala/org/apache/spark/storage/BlockManagerReplicationSuite.scala

```diff
@@ -494,7 +494,9 @@ class BlockManagerProactiveReplicationSuite extends BlockManagerReplicationBehav
 
     val newLocations = master.getLocations(blockId).toSet
     logInfo(s"New locations : $newLocations")
-    assert(newLocations.size === replicationFactor)
+    eventually(timeout(5 seconds), interval(10 millis)) {
+      assert(newLocations.size === replicationFactor)
```

Comment:
IMHO, we can not remove the first sleep. For example there are three blockmanager A, B, C. When we stats to remove BM-A, all blocks in BM-A will be replicated to BM-B and BM-C. We can not remove BM-B immediately or too fast, as there may be no enough time to do replication and new block info may can not be registered to master properly. So, we should instead give a little more time to sleep just like my fist fix. But it is OK to remove the second sleep. 
@kayousterhout Tell me if i was missing something.